modules/gitops-argo-cd-round-robin-sharding-algorithm.adoc

```diff
@@ -0,0 +1,138 @@
+// Module included in the following assemblies:
+//
+// * argocd_instance/argo-cd-cr-component-properties.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-argo-cd-round-robin-sharding-algorithm_{context}"]
+= Enabling the round-robin sharding algorithm
+
+:FeatureName: The `round-robin` sharding algorithm
+include::snippets/technology-preview.adoc[]
+
+By default the Argo CD Application Controller uses the non-uniform `legacy` hash-based sharding algorithm to assign clusters to shards. This can result in uneven cluster distribution. You can enable the `round-robin` sharding algorithm to achieve more equal cluster distribution across all shards. 
+
+Using the `round-robin` sharding algorithm in the {gitops-title} provides the following benefits:
+
+* Ensure more balanced workload distribution
+* Prevent shards from being overloaded or underutilized
+* Optimize the efficiency of computing resources
+* Reduce the risk of bottlenecks
+* Improve overall performance and reliability of the Argo CD system
+
+The introduction of alternative sharding algorithms allows for further customization based on specific use cases. You can select the algorithm that best aligns with your deployment needs, which results in greater flexibility and adaptability in diverse operational scenarios.
+
+.Prerequisites
+* You have installed the {gitops-title} Operator in your cluster.
+* You have access to the {OCP} web console.
+* You have access to the cluster with `cluster-admin` privileges.
+
+.Procedure
+
+You can enable the `round-robin` sharding algorithm by either using the {OCP} web console or the command-line interface.
+
+* To enable the `round-robin` algorithm using the {OCP} web console, complete the following steps:
+
+. In the *Administrator* perspective of the web console, go to *Operators* → *Installed Operators*.
+
+. Click *{gitops-title}* from the installed operators and go to the *Argo CD* tab.
+
+. Click the Argo CD instance where you want to enable the `round-robin` sharding algorithm, for example `openshift-gitops`.
+
+. Click the *YAML* tab and edit the yaml file:
++
+.Example Argo CD instance with round-robin sharding algorithm enabled
+[source,yaml]
+----
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: openshift-gitops
+  namespace: openshift-gitops
+spec:
+  controller:
+    processors: {}
+    sharding:
+      enabled: true <1>
+      replicas: 3 <2>
+    env: <3>
+      - name: ARGOCD_CONTROLLER_SHARDING_ALGORITHM 
+        value: round-robin
+----
+<1> Set the `sharding.enabled` parameter to `true` to enable sharding.
+<2> Set the number of replicas to the desired value, for example `3`.
+<3> Set the sharding algorithm to `round-robin`.
+
+. Click *Save* to save the changes.
```

Comment:
I have no idea why I did not think to include this. Thank you for pointing that out!  :+1:  :100:

Reply:
Hello @Srivaralakshmi i just figured out why I did not include this message initially.  When i was writing the procedure, I tested it on the different instance. The message shows only if we try to edit the `openshift-gitops` instance, with other instances, this message does not show. What do you think would be the best course of action in such a situation? I would maybe write it as a note that if the user is updating the `openshift-gitops` instance, that message would show and they would have to click **save** again. WDYT?

Reply:
As per our discussion, the SME confirmed that the installation type here does not matter, the important user case here is that sharding is more likely to be used in cluster-scope installation because users in cluster-scope installation are more likely to hit performances issues. This is acknowledged by the fact that we are adding this section under the declarative cluster configuration.